zaza/model.py

```diff
@@ -725,32 +732,40 @@ def one_agent_status(model, status):
             lambda: len(model.units) > 0
         )
         logging.info("Waiting for all units to be idle")
-        await model.block_until(
-            lambda: model.all_units_idle(), timeout=timeout)
-        for application in model.applications:
-            check_info = states.get(application, {})
-            for unit in model.applications[application].units:
-                logging.info("Checking workload status of {}".format(
-                    unit.entity_id))
-                await model.block_until(
-                    lambda: check_unit_workload_status(
-                        model,
-                        unit,
-                        check_info.get('workload-status', 'active')),
-                    timeout=timeout)
-                check_msg = check_info.get('workload-status-message')
-                logging.info("Checking workload status message of {}".format(
-                    unit.entity_id))
-                if check_msg is not None:
-                    prefixes = (check_msg)
-                else:
-                    prefixes = approved_message_prefixes
-                await model.block_until(
-                    lambda: check_unit_workload_status_message(
-                        model,
-                        unit,
-                        prefixes=prefixes),
-                    timeout=timeout)
+        try:
+            await model.block_until(
+                lambda: model.all_units_idle(), timeout=timeout)
+        except concurrent.futures._base.TimeoutError as e:
+            raise ModelTimeout("Zaza has timed out waiting on the model to "
+                               "reach idle state.")
+        try:
+            for application in model.applications:
```

Comment:
Should this be explicitly `model.applications.keys()`?

If so, would it be better to iterate over `k, v ... .items()`?

Reply:
model.applications is a type mapping that acts like both a list of objects and a dictionary. The usage is mixed throughout zaza and libjuju itself. Note this was just a white space change.

Let's discuss. If you truly want this changed. I'll do it.